tests/trainer/test_trainer.py

```diff
@@ -905,6 +906,15 @@ def __init__(self, **kwargs):
         TrainerSubclass(abcdefg='unknown_arg')
 
 
+@pytest.mark.parametrize('trainer_params', [
+    OmegaConf.create({'max_epochs': 1, 'gpus': 1}),
+    OmegaConf.create({'max_epochs': 1, 'gpus': [0]}),
+])
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="test requires GPU machine")
+def test_trainer_omegaconf(trainer_params):
+    _ = Trainer(**trainer_params)
```

Comment:
I personally prefer to just call the constructor in such cases (your call of course).
```suggestion
    Trainer(**trainer_params)
```